### PR TITLE
upgrade guide for 0.10.0

### PR DESCRIPTION
**Pull request by jazzyfresh**
## Todo
* [x] default update stanza of rolling upgrade deployments
* [x] add raft backwards compatibility section according to this comment https://github.com/hashicorp/nomad/pull/6250#pullrequestreview-283215580

### CODE REVIEW

**Comment by jazzyfresh** [comment]
There might be more to include here, feel free to update the description or push directly to this branch.

File: website/source/guides/upgrade/upgrade-specific.html.md [hunk 1]
```diff
@@ -15,6 +15,18 @@ details provided for their upgrades as a result of new features or changed
 behavior. This page is used to document those details separately from the
 standard upgrade flow.
 
+## Nomad 0.10.0
+
+### Deployments
+
+Nomad 0.10 enables rolling upgrade deployments for jobs by default,
```
Comment (preetapan) [inline website/source/guides/upgrade/upgrade-specific.html.md]: This should say "service" jobs here and everywhere else below.

File: website/source/guides/upgrade/upgrade-specific.html.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ details provided for their upgrades as a result of new features or changed
...
+adding a default update stanza when you create and update jobs
+that do not already have an update stanza present.
+This does not affect existing jobs with an update stanza.
+In pre-0.10.0 releases, jobs without an update stanza were deployed
+with destructive updates by default. To regain this behavior,
+see [`update stanza`][update-stanza].
```
Comment (jazzyfresh) [inline website/source/guides/upgrade/upgrade-specific.html.md]: This didn't render correctly

Comment (preetapan) [reply website/source/guides/upgrade/upgrade-specific.html.md]: @jazzyfresh you will need to add a link at the end for update stanza. See the end of this file for how the other URLS are defined.

File: website/source/guides/upgrade/upgrade-specific.html.md [hunk 3]
```diff
@@ -15,6 +15,32 @@ details provided for their upgrades as a result of new features or changed
...
+with destructive updates by default. To regain this behavior and
+disable deployments, set `max_parallel` to 0. 
+For more information, see [`update` stanza][update].
+
+### Raft 3
+Nomad 0.10 defaults to Raft 3 which includes [Autopilot](/guides/operations/autopilot.html), 
+operator-friendly automatic cluster management. To enable autopilot, all servers
+in a Nomad cluster must be running with Raft protocol version 3 or later.
+
+If existing servers are still on Raft 1, users will need to set the
```
Comment (preetapan) [inline website/source/guides/upgrade/upgrade-specific.html.md]: suggest replacing `Raft` with `raft protocol version` here and below..

File: website/source/guides/upgrade/upgrade-specific.html.md [hunk 4]
```diff
@@ -15,6 +15,32 @@ details provided for their upgrades as a result of new features or changed
...
 
+## Nomad 0.10.0
+
+### Deployments
+
+Nomad 0.10 enables rolling deployments for service jobs by default
+and adds a default update stanza when a service job is created or updated.
+This does not affect jobs with an update stanza.
+In pre-0.10.0 releases, service jobs without an update stanza were deployed
+with destructive updates by default. To regain this behavior and
```
Comment (notnoop) [inline website/source/guides/upgrade/upgrade-specific.html.md]: Not sure if we ever called it "destructive updates" before.  If not, we might tweak the language to explain behavior.  Something like:

> In pre-0.10 releases, when updating a a service job without update stanza, all existing allocations are stopped, while new allocations start up, and may cause a service degradation or an outage.  You can regain this disructive behavior by setting `maz_parallel` to 0.

You can do more wordsmitthing if you are up for it?

File: website/source/guides/upgrade/upgrade-specific.html.md (hunk 3 above)
Comment (notnoop) [inline website/source/guides/upgrade/upgrade-specific.html.md]: I'd avoid users, maybe `operators` or `you`, or use passive voice like rest of paragraph.

Comment (jazzyfresh) [reply website/source/guides/upgrade/upgrade-specific.html.md]: done

File: website/source/guides/upgrade/upgrade-specific.html.md [hunk 5]
```diff
@@ -15,6 +15,32 @@ details provided for their upgrades as a result of new features or changed
...
+disable deployments, set `max_parallel` to 0. 
+For more information, see [`update` stanza][update].
+
+### Raft 3
+Nomad 0.10 defaults to Raft 3 which includes [Autopilot](/guides/operations/autopilot.html), 
+operator-friendly automatic cluster management. To enable autopilot, all servers
+in a Nomad cluster must be running with Raft protocol version 3 or later.
```
[the remark this answers is not in the document]
Comment (jazzyfresh) [reply website/source/guides/upgrade/upgrade-specific.html.md]: done

**Pull request merged**